openstack-cloud/src/main/java/jenkins/plugins/openstack/compute/JCloudsSlave.java

```diff
@@ -108,7 +108,7 @@ public String getJvmOptions() {
      * @see CloudInstanceDefaults#DEFAULT_INSTANCE_RETENTION_TIME_IN_MINUTES
      */
     public int getRetentionTime() {
-        if (overrideRetentionTime > 0) {
```

Comment:
1. just !=0
2. add comment here, if retentionTime < 0 - the slave will be kept forever